### ISSUE TITLE
`UPDATE` with join relevant conditions

### ISSUE THREAD

**Issue by patricebender** [cap-js/cds-dbs#93]
```
UPDATE.entity(Books).data( { texts: [ { locale: 'es', descr: 'Òla'} ] } ).where(`genre.name = 201`).forSQL()
```
yields this, invalid, CQN:
```json5
Query {
  UPDATE: {
    entity: { ref: [ 'sap.capire.bookshop.Books' ], as: 'Books' },
    data: { texts: [ { locale: 'es', descr: 'Òla' } ] },
    where: [ { ref: [ 'genre', 'name' ] }, '=', { val: 201 } ],
    from: {
      join: 'left',
      args: [
        { ref: [ 'sap.capire.bookshop.Books' ], as: 'Books' },
        { ref: [ 'sap.capire.bookshop.Genres' ], as: 'genre' }
      ],
      on: [
        { ref: [ 'genre', 'ID' ] },
        '=',
        { ref: [ 'Books', 'genre_ID' ] }
      ]
    }
  }
}
```

this could probably be translated to an `UPDATE` with a `where exists` subquery, correlating both tables.

**Comment by patricebender** [comment]
decision from the db steering meeting on 10th july:
- turn such queries into `UPDATE` with `where exists`

**Comment by patricebender** [comment]
addendum: We do not understand the token stream such as a `where` clause, it is not easily possible to turn `UPDATE.entity('bookshop.Books').where(`author.name LIKE '%Bron%' or ( author.name LIKE '%King' and title = 'The Dark Tower') and stock >= 15`)` into a corresponding `where exists` subquery. For now, I would forbid path expressions in `UPDATE` statements, until we found a proper solution. @stewsk  FYI

**Comment by patricebender** [comment]
resolved in #97
